ctapipe/reco/tests/test_HillasReconstructor.py

```diff
@@ -72,154 +83,260 @@ def test_h_max_results():
+                frame=AltAz(),
+            )
+
+            geom_camera_frame = source.subarray.tel[tel_id].camera.geometry
+
+            # this could be done also out of this loop,
+            # but in case of real data each telescope would have a
+            # different telescope_pointing
+            geom_telescope_frame = geom_camera_frame.transform_to(
+                TelescopeFrame(telescope_pointing=telescope_pointings[tel_id])
+            )
 
             mask = tailcuts_clean(
-                geom, dl1.image, picture_thresh=10.0, boundary_thresh=5.0
+                geom_telescope_frame, dl1.image, picture_thresh=10.0, boundary_thresh=5.0
             )
 
             try:
-                moments = hillas_parameters(geom[mask], dl1.image[mask])
-                hillas_dict[tel_id] = moments
-            except HillasParameterizationError:
+                moments_camera_frame = hillas_parameters(
+                    geom_camera_frame[mask], dl1.image[mask]
+                )
+                moments_telescope_frame = hillas_parameters(
+                    geom_telescope_frame[mask], dl1.image[mask]
+                )
+
+                if (moments_camera_frame.width.value > 0) and (moments_telescope_frame.width.value > 0):
+                    event_camera_frame.dl1.tel[
+                        tel_id
+                    ].parameters.hillas = moments_camera_frame
+                    dl1.parameters.hillas = moments_telescope_frame
+
+                    hillas_dict_camera_frame[tel_id] = moments_camera_frame
+                    hillas_dict_telescope_frame[tel_id] = moments_telescope_frame
+                else:
+                    continue
+
+            except HillasParameterizationError as e:
+                print(e)
                 continue
 
-        # construct a dict only containing the last telescope events
-        # (#telescopes < 2)
-        hillas_dict_only_one_tel = dict()
-        hillas_dict_only_one_tel[tel_id] = hillas_dict[tel_id]
-        with pytest.raises(TooFewTelescopesException):
-            fit.predict(hillas_dict_only_one_tel, subarray, tel_azimuth, tel_altitude)
-
-        # construct a hillas dict with the width of the last event set to 0
-        # (any width == 0)
-        hillas_dict_zero_width = hillas_dict.copy()
-        hillas_dict_zero_width[tel_id]["width"] = 0 * u.m
-        with pytest.raises(InvalidWidthException):
-            fit.predict(hillas_dict_zero_width, subarray, tel_azimuth, tel_altitude)
-
-        # construct a hillas dict with the width of the last event set to np.nan
-        # (any width == nan)
-        hillas_dict_nan_width = hillas_dict.copy()
-        hillas_dict_zero_width[tel_id]["width"] = np.nan * u.m
-        with pytest.raises(InvalidWidthException):
-            fit.predict(hillas_dict_nan_width, subarray, tel_azimuth, tel_altitude)
+        if (len(hillas_dict_camera_frame) > 2) and (len(hillas_dict_telescope_frame) > 2):
```

Comment:
Maybe this is needed and I am missing something, but why would one dict have enough entries and the other one does not?

Reply:
You're right, here it is probably OP since the only cut is width>0....
though the reconstruction is done in the same way from both frames, so it is coherent that the check is done from both inputs